### ISSUE TITLE
Graph Editor adds edges (when lightweight edges are used) to the schema!

### ISSUE THREAD

**Issue by pmeisen** [orientechnologies/orientdb#6465]
## OrientDB Version, operating system, or hardware.
- v2.0 SNAPSHOT[ ] - .18[ ] .17[ ] .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.1 SNAPSHOT[ ] - .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.2.4 [x]
## Operating System
- [ ] Linux
- [x] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 
## Expected behavior and actual behavior

The Graph Editor (web-UI) adds an edge class to the databases's schema, if following an IN or OUT edge in the Graph Editor and if lightweight edges are used.

The Graph Editor should never modify the schema of the database, that is a no go, even if it's just adding an unused class.
## Steps to reproduce the problem
1. open a database using lightweight edges
2. add a vertex to the database (e.g., of class V1)
3. add another vertex  (e.g., of class V2) and connect the two added vertex with a lightweight edge (labeling TEST)
4. open the web ui and open the Graph Editor
5. SELECT \* FROM V1
6. click the icon to follow the IN or OUT connection and select the label TEST
7. the second vertex appears (really nice 👍 ) and the edge is shown
8. click on the schema tab and find a edge class added called TEST 👎

**Comment by maggiolo00** [comment]
@pmeisen 

where did you configure lightweight edges?

How did you create the connection from v1 and v2 with label Test/

**Comment by pmeisen** [comment]
@maggiolo00 

Prior to adding the vertices and the edges in the Java Application:
`factory.setUseLightweightEdges(true);`

The problem is not that the edges are not lightweight, they are... the problem is that the web UI does add the edge-classes, which it definitely should not do... should it?!

**Comment by maggiolo00** [comment]
@pmeisen 

Studio does not create any edge-class directly.

When you expand a relationship from the Web UI, this is the query executed.

```SQL
select expand(unionAll(outE('Test'),out('Test')) )  from #22:0
```
change  #22:0 with the your rid

This seems to create the edge class if not exists.

**Comment by pmeisen** [comment]
First of all, I can confirm that executing the statement via the console also creates the edge class!

Ok, but  isn't it a bug that a **SELECT statement modifies the schema** of a database? If that is wanted, I honestly don't see any chance to use such a system in production! So **could you confirm that this is a bug and that it will be fixed**? Is there any permission that will avoid the changes?

Just noticed another thing:
If I have multiple edges and I click once on the show edges, it creates the class. If I execute the same query again (for the same or any other vertex), I get:

```java.lang.IllegalArgumentException: Type error. The class TEST does not extend class 'E' and therefore cannot be considered an Edge```

**Comment by maggiolo00** [comment]
hi @pmeisen 

i think that is definitely a bug. A select should not change the schema. 
The class should be created only when the lightweight is converted to Regular edges.

**Comment by tglman** [comment]
hi All,

I was already aware of this issue, the reason behind this behaviour is that for work properly the orientdb engine need the classes for all the edges defined, as early fix we introduced the creation of the class in case is missed, this brought this not consistent behaviour, we are working to redesign some part in 3.0.

I would suggest to ensure that you have all the classes for all the edges to avoid this behaviour.

we are investigating if we can avoid this in 2.2.x, but is quite unluckely.

Regards

**Comment by pmeisen** [comment]
If I add the classes, I get the error (when firing the select) as mentioned in my previous post.

```
java.lang.IllegalArgumentException: Type error. The class TEST does not extend class 'E' and therefore cannot be considered an Edge
```

The documentation states, that lightweight edges are recommended, because of performance reasons. Thus, I just don't understand the comment @tglman . Does that mean, you cannot use lightweight edges until 3.0? Or can you use lightweight edges and you just cannot use 

```select expand(unionAll(outE('Test'),out('Test')) )  from #22:0```

Well, I think I just make the UI unavailable and make don't let anyone query the database and only offer pre-selected, pre-tested queries. Or should I remove the lightweight edges? What is the recommendation to do?

**Comment by smolinari** [comment]
As @tglman said, simply make sure the edge classes are available to begin with. Or put another way, don't query with imaginative edges. :smile: (It is a poor answer, yes.)

@tglman, I personally don't get the logic of creating classes with a select for the ODB engine to work properly. I would have imagined, if someone queries with a missing edge class (like misspelled!), the reply from ODB should simply be that the edge class doesn't exist. Even lightweight edges have their name posted in the schema, correct? 

In other words, this behavior is completely unacceptable. 

Scott

**Comment by pmeisen** [comment]
Thanks @smolinari, 

1st of all for the solution (which is still kind of - well, I have to copy all the data from one schema to another and make sure that the labels used for lightweight edges exist as a edge class, which is not used... anyways) - thanks for that!

2nd of all for the last sentence 👍  - no not your name, the one before that :)

**Comment by tglman** [comment]
Hi all,

@smolinari if the class not exist in data it would not be created, is only for classes referred in data.

@pmeisen The solution is to create all the needed edge classes ahead also if you have lightweigth edges, if right now you have an class that not extend `E`  just drop it and recreate it extending `E`.

Hope it help

**Comment by smolinari** [comment]
@tglman - I'll be honest, I don't understand. But if @pmeisen is happy, then I am too. :smile: 

Scott

**Comment by pmeisen** [comment]
@smolinari Happy?! Far away from that... but I can code a nice workaround now, which I have to track and make sure when we update to be removed. I love stuff like that, makes my life so much easier *ironie*!

Anyways, cannot wait to get this bug fixed and update to the hopefully "bug-free" version 3.x.x one day :)!
